api/cluster.go

```diff
@@ -96,7 +99,12 @@ func (s *Server) getData(ctx *middleware.Context, request models.GetData) {
 func (s *Server) indexDelete(ctx *middleware.Context, req models.IndexDelete) {
 	defs, err := s.MetricIndex.Delete(req.OrgId, req.Query)
 	if err != nil {
-		response.Write(ctx, response.NewError(http.StatusBadRequest, err))
+		// errors can be due to bad user input or corrupt index.
+		if strings.Contains(err.Error(), "Index is corrupt") {
+			response.Write(ctx, response.WrapError(err))
+		} else {
+			response.Write(ctx, response.NewError(http.StatusBadRequest, err.Error()))
+		}
```

Comment:
this, and some of the other error changes here looks like an anti-pattern to me. specifically:
1) doing string checks on error messages
2) coupling logic in a caller directly to internals and implementation details of the callee.

I think this will be hard to maintain and easily result in assumptions that become out of date when code being called changes.

It seems there's really only 2 classes of errors:
A) user did something wrong (bad "request" in an abstract sence. could be http request from a user, or CLI input)
B) internally, something went wrong. could be a network timeout talking to peers, a slice out of bounds, etc.

I think for B, in most deployments (those for "regular users" e.g. the ones we run for worldping and hosted metrics) we don't want to communicate the problem to the user. it's critical that MT logs the detailed error and the operator can take it from there (using ELK or other means of monitoring). I think we can probably just tell the user "internal (server) error" in all cases without any detail whatsoever.

In those deployments where all users are also operators (e.g. our internal stack that we use for monitoring) it may be convenient to return the concrete error back to the user, however, since we already established that there should be good tooling/monitoring to act upon errors logged in the logfile, we shouldn't go out of our way to implement this.

So basically, I think the only thing we need is:
- a very simple Error type that has a bool flag to denote whether it was internal or due to user and a message field
- internal errors can be logged where they happen, or at a higher level. we should retain some flexibility here.
- if the error was internal, we just tell users "internal error" (or "internal server error" for the api server with code 500)

something like https://github.com/davecheney/errors looks neat in that you can add context to an error as you pass through different functions. it adresses the anti pattern of getting the string msg from an error, prefixing it with context, and returning it as a different error (and potentially different error type), but it could be more than what we need

@woodsaj thoughts?